patchwork-rendering/src/main/java/net/minecraftforge/client/event/GuiScreenEvent.java

```diff
@@ -0,0 +1,751 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import net.minecraftforge.eventbus.api.Event;
+import org.lwjgl.glfw.GLFW;
+
+import net.minecraft.client.gui.Element;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.ingame.AbstractInventoryScreen;
+import net.minecraft.client.gui.widget.AbstractButtonWidget;
+import net.minecraft.client.gui.widget.ButtonWidget;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Event classes for GuiScreen events.
+ *
+ * @author bspkrs
+ */
+@Environment(EnvType.CLIENT)
+public class GuiScreenEvent extends Event {
+	private final Screen gui;
+
+	public GuiScreenEvent() {
```

Comment:
The EventBus constructors should have a comment that says `//For EventBus` above those constructors, and calling `super()` is preferred over `this(null)`

Reply:
the `gui` is final, it needs to be initialised. `this(null)` should cause no issues.

Reply:
If you're sure, go ahead.